usaspending_api/etl/elasticsearch_loader_helpers/delete_data.py

```diff
@@ -115,73 +72,171 @@ def delete_docs_by_unique_key(client: Elasticsearch, key: str, value_list: list,
     if not index:
         raise RuntimeError("index name must be provided")
 
+    if not _is_allowed_key_field_type(client, key, index):
+        msg = (
+            f'Cannot perform deletes in index "{index}" by key field "{key}" because its type is not one of '
+            f"the allowed field types, or the field was not found in that index."
+        )
+        logger.error(format_log(msg=msg, action="Delete", name=task_id))
+        raise RuntimeError(msg)
+
+    if delete_chunk_size > 65536:
+        # 65,536 is max number of terms that can be added to an ES terms filter query
+        msg = (
+            f"{delete_chunk_size} is greater than 65,536, which is the max number of terms that can be added to an ES "
+            f"terms filter query"
+        )
+        logger.error(format_log(msg=msg, action="Delete"))
+        raise RuntimeError(msg)
+
+    chunks_processed = 0
     deleted = 0
     is_error = False
     try:
-        # 65,536 is max number of terms that can be added to an ES terms filter query
-        values_generator = chunks(value_list, 50000)
+        values_generator = chunks(value_list, delete_chunk_size)
         for chunk_of_values in values_generator:
-            # Creates an Elasticsearch query criteria for the _delete_by_query call
-            q = ES_Q("terms", **{key: chunk_of_values})
             # Invoking _delete_by_query as per the elasticsearch-dsl docs:
             #   https://elasticsearch-dsl.readthedocs.io/en/latest/search_dsl.html#delete-by-query
-            response = Search(using=client, index=index).filter(q).delete()
-            chunk_deletes = response["deleted"]
-            deleted += chunk_deletes
+            # _refresh is deferred til the end of chunk processing
```

Comment:
Recommend replacing with `until` or `'til`

```suggestion
            # _refresh is deferred until the end of chunk processing
```